presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -198,7 +200,7 @@ static double getSourceTablesSizeInBytes(PlanNode node, Lookup lookup, StatsProv
         }
 
         List<PlanNode> sourceNodes = PlanNodeSearcher.searchFrom(node, lookup)
-                .whereIsInstanceOfAny(ImmutableList.of(TableScanNode.class, ValuesNode.class, RemoteSourceNode.class))
+                .whereIsInstanceOfAny(ImmutableList.of(TableScanNode.class, ValuesNode.class, RemoteSourceNode.class, CteProducerNode.class, CteConsumerNode.class))
```

Comment:
Why CteProducerNode is also source nodes?

Reply:
Nice catch, yes unintended